byway/src/main/webapp/track-interests.js

```diff
@@ -0,0 +1,55 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+let selected = [];
+
+/** 
+ * Track the selected interests in an array according to their status. 
+ * @param elem tracks the current button element chosen.
+ */
+function checkStatus(elem) {
+  if(elem.className === "btnActive") {
+    selected.push(elem.innerText);
+  } else {
+    selected = selected.filter(function(interest) {
+      return interest !== elem.innerText;
+    });
+  }
+}
+
+/** 
+ * Update the class of the buttons to change their style.
+ * @param elem tracks the current button element chosen.
+ */
+function switchStatus(elem) {
+  let switchClass = (elem.className === "btn"? "btnActive": "btn");
+  elem.className = switchClass;
+  checkStatus(elem);
+}
+
+/** Display all the buttons onscreen with independent onClick events. */
+function loadButtons() {
+  fetch('/places')
+  .then(response => response.json())
+  .then((places) => {
+    let buttonSection = document.getElementById("interests");
+    places.forEach((place) => {
+      let button = document.createElement("button");
+      button.innerText = place;
+      button.setAttribute("onClick", "switchStatus(this)");
+      button.setAttribute("class", "btn");
```

Comment:
Be consistent about whether you're using the Element properties or `setAttribute`. Personally I think `setAttribute` is more likely to result in mistakes, since the name of the property you're setting ends up as a field inside the `setAttribute` call instead of being an actual javascript field reference, so there's more to have to read through to check that it's right as a human, and possibly automated tools may be better able to help you when using the special properties.

Reply:
I tried using it the other way by setting the attribute. I called `button.onClick = switchStatus(this)` and others but they did properly set. Some did, like `.innerText`, and others didn't, so that is why it is a bit inconsistent.

Reply:
Fixed

Reply:
Note that if you wrote `button.onClick = switchStatus(this)` you were assigning `onClick` to the *result* of *calling* `switchStatus`. Remember, when you assign a function call to something, you are assigning the return value of the function. Event handlers are not special in this regard. If you want to assign a function you need to provide an expression that evaluates to a function, typically either something like `switchStatus` (note: no parenthesis, so not calling), or a function expression, like `() => switchStatus(button)`.